### PR TITLE
Add a catalog task to run the OpenSSF Security Scorecard project.

### PR DESCRIPTION
**Pull request by dlorenc**
# Changes

This task is very simple right now, but will be extended to support
other flags and output formats int he future.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [X] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [X] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [X] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [X] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [X] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [X] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [X] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [X] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

File: task/scorecard/0.1/scorecard.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    - name: REPO
+      description: The repository URL to inspect
+      type: string
+    - name: GITHUB-OAUTH_SECRET
+      description: The GitHub secret with credentials
+      type: string
+      default: "github"
+  steps:
+    - name: scorecard
+      image: gcr.io/openssf/scorecard:latest
```
Comment (vinamra28) [inline task/scorecard/0.1/scorecard.yaml:27]: can we change the `latest` tag to `tag+digest` format

Comment (dlorenc) [reply task/scorecard/0.1/scorecard.yaml:27]: Done!

File: task/scorecard/0.1/scorecard.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    - name: REPO
+      description: The repository URL to inspect
+      type: string
+    - name: GITHUB-OAUTH_SECRET
+      description: The GitHub secret with credentials
+      type: string
+      default: "github"
+  steps:
+    - name: scorecard
+      image: gcr.io/openssf/scorecard:latest+sha256:c16e433f68799fe69dc0f9675c7ba5d2553846b57262550c86c71165269a7961
```
Comment (vinamra28) [inline task/scorecard/0.1/scorecard.yaml:27]: my bad :man_facepalming: 
```suggestion
      image: gcr.io/openssf/scorecard@sha256:c16e433f68799fe69dc0f9675c7ba5d2553846b57262550c86c71165269a7961
```

Comment (vdemeester) [reply task/scorecard/0.1/scorecard.yaml:27]: @dlorenc is `sha256:c16e433f68799fe69dc0f9675c7ba5d2553846b57262550c86c71165269a7961` pointing to `:latest` ? If yes, the digest might become unreliable next time something is pushed on `:latest` and gcr does a garbage collection.

Comment (dlorenc) [reply task/scorecard/0.1/scorecard.yaml]: All set!

**Review by vinamra28 — COMMENTED** [review]
Looks good. Thanks
/lgtm

**Pull request merged**